deployers/manifestreader.go

```diff
@@ -131,7 +131,13 @@ func (deployer *ManifestReader) HandleYaml(sdeployer *ServiceDeployer, manifestP
 }
 
 func (reader *ManifestReader) SetDependencies(deps map[string]utils.DependencyRecord) error {
+
+	d := reader.serviceDeployer
```

Comment:
Trying to understand why we need a mutex semaphore... where do/can we have multiple methods accessing service deployer at the same time?

Reply:
@pritidesai scanning the code, I do not see where we are attempting anything multi-threaded where we would be subject to the same servicedeployer being accessed by more than one thread.  In fact, it seems the only place I saw where we attempted threads is in what looks like incomplete code in report.go.

Reply:
@mrutkows I am trying to justify as well but I have seen issues when a manifest has specified multiple dependencies on the same package, clone dependency failing somewhere when one invocation downloads the dependency and deploys it and deletes the downloaded zip file where the other invocation fails with missing downloaded zip file, there is a somewhere race condition. But I think I will remove all locking unlocking and update the PR , will address concurrency in separate PR.

Reply:
@pritidesai yes, let's not lock until we understand the problem.  If we need to add more trace in order to visualize the problem (and add new tests), please let us do that first.